src/common/transformations/src/transformations/smart_reshape/lstm_states_broadcast.cpp

```diff
@@ -0,0 +1,171 @@
+        return false;
+
+    const auto& constant_copy = constant_state->copy_with_new_inputs({});
+    const auto& broadcast_by_batch = make_shared<Broadcast>(
+        constant_copy,
+        make_shared<Concat>(ngraph::NodeVector{batch_delivering_node,
+                                               ngraph::op::util::make_try_fold<Gather>(
+                                                   ngraph::op::util::make_try_fold<ShapeOf>(constant_copy),
+                                                   Constant::create(ov::element::i64, ov::Shape{1}, {1}),
+                                                   Constant::create(ov::element::i64, ov::Shape{}, {0}))},
+                            0));
+    input.replace_source_output(broadcast_by_batch->output(0));
+    return true;
+}
+
+bool relax_batch_for_initial_states_of_lstm_in_ti(const shared_ptr<TensorIterator>& ti,
+                                                  const shared_ptr<LSTMCell>& lstm_cell) {
+    bool rewritten = false;
+    auto batch_delivering_node = deduce_outer_source_of_batch_for_inner_lstm_cell(ti, lstm_cell);
+    if (batch_delivering_node == nullptr)
+        return rewritten;
+    if (auto init_hidden_state = dynamic_pointer_cast<Parameter>(lstm_cell->get_input_node_shared_ptr(1))) {
+        auto outer_init_hidden_state_input = get_outer_input_of_ti_by_parameter(init_hidden_state, ti);
+        rewritten |= broadcast_state_by_batch(outer_init_hidden_state_input, batch_delivering_node);
+    }
+    if (auto init_cell_state = dynamic_pointer_cast<Parameter>(lstm_cell->get_input_node_shared_ptr(2))) {
+        auto outer_init_cell_state_input = get_outer_input_of_ti_by_parameter(init_cell_state, ti);
+        rewritten |= broadcast_state_by_batch(outer_init_cell_state_input, batch_delivering_node);
+    }
+    return rewritten;
+}
+
+bool relax_batch_for_initial_states_of_lstm(const shared_ptr<LSTMCell>& lstm_cell) {
+    bool rewritten = false;
+    const auto& batched_shape = make_shared<ShapeOf>(lstm_cell->get_input_source_output(0));
+    const auto& batch_delivering_node = make_shared<Gather>(batched_shape,
+                                                            Constant::create(ov::element::i64, ov::Shape{1}, {0}),
+                                                            Constant::create(ov::element::i64, ov::Shape{}, {0}));
+    rewritten |= broadcast_state_by_batch(lstm_cell->input(1), batch_delivering_node);
+    rewritten |= broadcast_state_by_batch(lstm_cell->input(2), batch_delivering_node);
+    return rewritten;
+}
+
+bool ov::pass::LSTMStatesBroadcast::run_on_model(const shared_ptr<ov::Model>& f) {
+    RUN_ON_FUNCTION_SCOPE(LSTMStatesBroadcast);
+    bool rewritten = false;
+    for (auto& node : f->get_ordered_ops()) {
+        // Recursively apply transformation for sub-graph based operations
+        if (const auto& sub_graph_node = dynamic_pointer_cast<ov::op::util::SubGraphOp>(node))
```

Comment:
SubGraphOp -> MultiSubgraphOp, e.g. If op
We also need to run this transformation for all bodies

Reply:
Let's leave it for the time we have a real model with such case.